### ISSUE TITLE
Improving web accessibility

### ISSUE THREAD

**Issue by ashleytqy** [jupyter/notebook#3471]
Running `pa11y` (https://github.com/pa11y/pa11y) on `http://localhost:8888/tree` gives these errors:

```
Welcome to Pa11y

 > Running Pa11y on URL http://localhost:8888/tree

Results for URL: http://localhost:8888/login?next=%2Ftree

 • Error: The html element should have a lang or xml:lang attribute which describes the language of the document.
   ├── WCAG2AA.Principle3.Guideline3_1.3_1_1.H57.2
   ├── html
   └── <html><head> <meta charset="utf-8...</html>

 • Error: This password input element does not have a name available to an accessibility API. Valid names are: label element, title attribute, aria-label attribute, aria-labelledby attribute.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.InputPassword.Name
   ├── #password_input
   └── <input type="password" name="password" id="password_input" class="form-control">

 • Error: This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrapped around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.F68
   ├── #password_input
   └── <input type="password" name="password" id="password_input" class="form-control">

 • Error: Duplicate id attribute value "password_input" found on the web page.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
   ├── #password_input
   └── <input type="password" name="password" id="password_input" class="form-control" placeholder="Token">

 • Error: This password input element does not have a name available to an accessibility API. Valid names are: label element, title attribute, aria-label attribute, aria-labelledby attribute.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.InputPassword.Name
   ├── #password_input
   └── <input type="password" name="password" id="password_input" class="form-control" placeholder="Token">

 • Error: This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrapped around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.F68
   ├── #password_input
   └── <input type="password" name="password" id="password_input" class="form-control" placeholder="Token">

 • Error: This password input element does not have a name available to an accessibility API. Valid names are: label element, title attribute, aria-label attribute, aria-labelledby attribute.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.InputPassword.Name
   ├── #new_password_input
   └── <input type="password" name="new_password" id="new_password_input" class="form-control" placeholder="New password" required="">

 • Error: This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrapped around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
   ├── WCAG2AA.Principle1.Guideline1_3.1_3_1.F68
   ├── #new_password_input
   └── <input type="password" name="new_password" id="new_password_input" class="form-control" placeholder="New password" required="">

 • Error: Duplicate id attribute value "login_submit" found on the web page.
   ├── WCAG2AA.Principle4.Guideline4_1.4_1_1.F77
   ├── #login_submit
   └── <button type="submit" id="login_submit" class="btn btn-default">Log in and set new password</button>

9 Errors
```
Suggested fix: making changes as per the output above

**Comment by ckilcrease** [comment]
Hi, could I work on this issue/making these changes?

**Comment by takluyver** [comment]
Certainly! Can you see where to start?

**Comment by ckilcrease** [comment]
@takluyver I believe so; I'd be working on the html files in the notebook/templates directory, and making the suggested changes/additions to address the issues that come up when running pa11y (in accordance with the Web Content Accessibility guidelines) - correct?

**Comment by takluyver** [comment]
Yep, that sounds good. The language one might be tricky, because in
principle the interface can be translated, though in practice we haven't
made that easy yet. But I think all the other points should be
straightforward to fix.

**Comment by ckilcrease** [comment]
Great, I’ll get started today!

**Comment by craigbrett17** [comment]
Is this related to #1801? 

I'm trying to use notebook with NVDA and it's... hit and miss to put it mildly. 

Happy to help, I know a thing or two about this sort of thing. Though this issue is closed, so maybe the fixes are on the way already

**Comment by takluyver** [comment]
Yup, #1801 is the more general issue, this is about some specific problems identified by one tool, which are (hopefully) fixed in a recent pull request. Those changes aren't released yet - they'll come as part of notebook 5.5, which will hopefully be soon.

If you're interested in testing the development version with these changes, or working on further accessibility improvements, have a look at the docs on [setting up a development environment](http://jupyter-notebook.readthedocs.io/en/stable/contributing.html#setting-up-a-development-environment).
